gradoop-common/src/main/java/org/gradoop/common/model/impl/properties/strategies/PropertyValueStrategyFactory.java

```diff
@@ -0,0 +1,191 @@
+  }
+
+  /**
+   * Get a strategy which corresponds the provided class. If there is no mapping for the provided
+   * class in the class-strategy map, or the value of the parameter is {@code null}, an instance of
+   * {@code NoopPropertyValue} is returned.
+   *
+   * @param c some class
+   * @return strategy class which is able to handle the provided type.
+   */
+  public static PropertyValueStrategy get(Class c) {
+    PropertyValueStrategy strategy = INSTANCE.classStrategyMap.get(c);
+    if (strategy == null) {
+      if (Map.class.isAssignableFrom(c)) {
+        strategy = INSTANCE.classStrategyMap.get(Map.class);
+      } else if (Set.class.isAssignableFrom(c)) {
+        strategy = INSTANCE.classStrategyMap.get(Set.class);
+      } else if (List.class.isAssignableFrom(c)) {
+        strategy = INSTANCE.classStrategyMap.get(List.class);
+      }
+    }
+    return strategy == null ? INSTANCE.noopPropertyValueStrategy : strategy;
+  }
+
+  /**
+   * Returns value which is represented by the the provided byte array. Assumes that the value is
+   * serialized according to the {@code PropertyValue} standard.
+   *
+   * @param bytes byte array of raw bytes.
+   * @return Object which is the result of the deserialization.
+   */
+  public static Object fromRawBytes(byte[] bytes) {
+    PropertyValueStrategy strategy = INSTANCE.byteStrategyMap.get(bytes[0]);
+    return strategy == null ? null : strategy.get(bytes);
+  }
+
+  /**
+   * Compares two values.
+   *
+   * @param value first value.
+   * @param other second value.
+   * @return a negative integer, zero, or a positive integer as {@code value} is less than, equal
+   * to, or greater than {@code other}.
+   */
+  public static int compare(Object value, Object other) {
+    if (value != null) {
+      PropertyValueStrategy strategy = get(value.getClass());
+      return strategy.compare(value, other);
+    }
+
+    return 0;
+  }
+
+  /**
+   * Get byte array representation of the provided object. The object is serialized according to the
+   * {@code PropertyValue} standard.
+   *
+   * @param value to be serialized.
+   * @return byte array representation of the provided object.
+   */
+  public static byte[] getRawBytes(Object value) {
+    if (value != null) {
+      return get(value.getClass()).getRawBytes(value);
+    }
+    return new byte[] {0};
+  }
+
+  /**
+   * Returns strategy mapping to the provided byte value.
+   *
+   * @param value representing a data type.
+   * @return strategy class.
+   */
+  public static PropertyValueStrategy get(byte value) {
+    return INSTANCE.byteStrategyMap.get(value);
+  }
+
+  /**
+   * Get strategy by object.
+   *
+   * @param value some object.
+   * @return strategy that handles operations on the provided object type or
+   * {@code NoopPropertyValueStrategy} if no mapping for the given type exists.
+   */
+  public static PropertyValueStrategy get(Object value) {
+    if (value != null) {
+      return get(value.getClass());
```

Comment:
This can return `null` for unsupported types, please check for that.

Reply:
if the type is unsupported, a NoopPropertyValueStrategy will be returned. I will change that since i dont quite understand if that is reasonable behaviour.

Reply:
Does NoopPropertyValue still exist?